source/k64f_eth_nanostack_port.c

```diff
@@ -444,45 +292,57 @@ static int8_t k64f_eth_send(uint8_t *data_ptr, uint16_t data_len)
     }
 
-    uint8_t *aligned_ptr = (uint8_t*) ENET_ALIGN((uint32_t )buf_ptr,
-                                                 TX_BUF_ALIGNMENT);
-
-    buf_desc_ring->tx_data_buf_ptr[index] = buf_ptr;
-
+    // Protect against reclaim routine swallowing tx_buf before we set READY
+    eth_if_lock();
+    base_DS.tx_buf[index] = buf_ptr;
+    uint8_t *aligned_ptr = (uint8_t*) ENET_ALIGN(buf_ptr, TX_BUF_ALIGNMENT);
     memcpy(aligned_ptr, data_ptr, data_len);
 
-    //if (aligned_ptr[12] == 0x86 && aligned_ptr[13] == 0xDD) {
-    //    tr_debug("Data TX %u %s", data_len, trace_array(aligned_ptr, 24));
-    //}
-
-    k64f_update_txbds(buf_desc_ring, index, aligned_ptr, data_len, 1);
-    buf_desc_ring->txb_aligned[index] = NULL;
-
-    index = (index + 1) % ENET_TX_RING_LEN;
-    buf_desc_ring->tx_buf_free_index = index;
-    //tr_debug("Next TX buffer = %i", index);
+    /* Setup transfers */
+    global_enet_handle.txBdCurrent->buffer = aligned_ptr;
+    global_enet_handle.txBdCurrent->length = data_len;
+    global_enet_handle.txBdCurrent->control |=
+            (ENET_BUFFDESCRIPTOR_TX_READY_MASK
+                    | ENET_BUFFDESCRIPTOR_TX_LAST_MASK);
+
+    /* Increase the buffer descriptor address. */
+    if (global_enet_handle.txBdCurrent->control
+            & ENET_BUFFDESCRIPTOR_TX_WRAP_MASK) {
+        global_enet_handle.txBdCurrent = global_enet_handle.txBdBase;
+    } else {
+        global_enet_handle.txBdCurrent++;
+    }
+    eth_if_unlock();
 
-    enet_hal_active_txbd(BOARD_DEBUG_ENET_INSTANCE_ADDR);
+    /* Active the transmit buffer descriptor. */
+    ENET->TDAR = ENET_TDAR_TDAR_MASK;
 
     return 0;
 }
 
-/* This function is calle by arm_eth_phy_k64f_address_write() which is in turn
- * called by nanostack in order to set up MAC address */
+
+/** \brief  Sets up MAC address
+ *
+ * This function is calle by arm_eth_phy_k64f_address_write() which is in turn
+ * called by nanostack in order to set up MAC address
+ *
+ *  \param[in] address_ptr   Pointer to MAC address block
+ */
 static void k64f_eth_set_address(uint8_t *address_ptr)
 {
-    enet_mac_config_t*mac_config =
-            &(ethernet_mac_config[BOARD_DEBUG_ENET_INSTANCE]);
-
     /* When pointer to the MAC address is given. It could be 48-bit EUI generated
      * from Radio, like atmel RF or manually inserted. Preferred method.*/
-    memcpy(mac_config->macAddr, address_ptr, kEnetMacAddrLen);
-    enet_hal_set_mac_address(device_id, mac_config->macAddr);
+    ENET_SetMacAddr(ENET, address_ptr);
 }
 
-/* This function sets the MAC address and its type for the Ethernet interface*/
-/* Only type supported is 48 bits. If address_ptr is NULL, we will try to set the
- * address using mbed-drivers. */
+
+/** \brief  Stack sets the MAC address using this routine
+ *
+ * This function sets the MAC address and its type for the Ethernet interface* Only type supported is 48 bits.
+ *  \param[in] address_type   Type of MAC address, i.e., 48 bit, 64 bit etc.
```

Comment:
If you're adding doxygen, make it accurate - missing parameter and missing linewrap. "Successful"